src/lsp-server.ts

```diff
@@ -243,6 +257,26 @@ export class LspServer {
         return rootPath;
     }
 
+    private findClosestTag(tags: any[], fileName: string, line: number) {
+        // Priority for shoosing tags (high to low):
+        // 1. if line number equal or less than given line
+        // 2. absolute distance between two lines
+        // 3. if in the same file
+        const inFileTags = tags.filter(tag => tag.file = fileName);
+        if (inFileTags.length !== 0) {
+            inFileTags.sort(function(l, r) {
+                return (Math.abs(l.lineNumber - line) - Math.abs(r.lineNumber - line));
+            });
+            inFileTags.forEach(tag => {
+                if (tag.lineNumber <= line) {
```

Comment:
Seems we just consider the tags ahead of the hover line, there is no need to sort the lines behind the hover line, right?